sort/pigeonhole_sort.dart

```diff
@@ -0,0 +1,42 @@
+void pigeonholeSort(List arr) {
```

Comment:
Would you mind to add some tests?
```suggestion
import 'package:test/test.dart';

void pigeonholeSort(List arr) {
```